src/components/Article.jsx

```diff
@@ -35,9 +44,54 @@ const {articleId} = useParams();
         }) 
     }, [articleId])
 
-useEffect(() => {
 
-})    
+//voting functions
+const upVote = () => {
+   setUserVote((currVote) => {
+    if (currVote === 1) {
+        setVoteMessage("You can only upvote once!")
+        return 1
+    } else if (currVote === -1) {
+        setVoteMessage("Your vote has been reset to neutral, click again to upvote")
+    } else {
+        setVoteMessage("Thanks for giving your feedback!")
+    }
+
+    changeArticleVote(articleId, 1)
+    .catch(() => {
+        setVoteMessage("Oops, your vote hasn't worked, try again")
+        return currVote
+    })
+
+    return ++currVote;
+   })
+}
+
+const downVote = () => {
+    setUserVote((currVote) => {
+     if (currVote === -1) {
+         setVoteMessage("You can only downvote once!")
+         return -1
+     } else if (currVote === 1) {
+         setVoteMessage("Your vote has been reset to neutral, click again to downvote")
+     } else {
+         setVoteMessage("Thanks for giving your feedback!")
+     }
+
+     changeArticleVote(articleId, -1)
+     .catch(() => {
+        setVoteMessage("Oops, your vote hasn't worked, try again")
+        return currVote
+    })
+
+     return --currVote;
```

Comment:
Because `currVote` is a primitive value, this shouldn't have any practical downsides, but nonetheless you ought to avoid mutating any state values. If you're in good habits, then you're less likely to make the mistakes later on when it does matter.

Reply:
fair point, have gone for the more pedestrian - return currVote -1